src/platform/plugins/shared/data/server/search/strategies/ese_search/ese_search_strategy.ts

```diff
@@ -166,12 +166,20 @@ export const enhancedEsSearchStrategyProvider = (
       throw new KbnSearchError(`"params.index" is required when performing a rollup search`, 400);
     }
 
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { ignore_unavailable, preference, ...params } = {
+      ...querystring,
+      ...request.params,
+      index: request.params.index,
+    };
+
     try {
       const esResponse = await client.rollup.rollupSearch(
         {
-          ...querystring,
-          ...request.params,
-          index: request.params.index,
+          ...params,
+          // Not defined in the spec, and the client places it in the body.
+          // This workaround allows us to force it as a query parameter.
+          querystring: { ...params.querystring, ignore_unavailable, preference },
```

Comment:
It doesn't seem to matter, but it looks like prior to this change, `timeout` and `track_total_hits` were being sent in the query string, but after this change they're being sent in the body. I believe both are accepted but just wanted to note this.

Reply:
Many Elasticsearch APIs accept the same parameters as query parameters or in the body, one reason being that by default, query strings are limited to 4096 characters. Clients prefer the body in this case, and I know the JS client changed the way it works in 9.0.0-alpha.4: https://github.com/elastic/elasticsearch-js/issues/2584#issuecomment-2685865679.

So this is likely fine and expected. What API is this though? [rollup_search](https://github.com/elastic/elasticsearch-specification/blob/main/specification/rollup/rollup_search/RollupSearchRequest.ts) does not seem to include `timeout` or `track_total_hits` parameters.